# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/cookiestore/CookieStoreParent.h
# Commit: 56cce39fb13c
# Full Hash: 56cce39fb13cd0623644bf9d40cbcec3473a93d1
# Author: edgul <edgul@mozilla.com>
# Date: 2025-06-03 21:32:22
# Description:
#   Bug 1968495 - Pass NotNull nsURIs across cookiestore IPC. r=baku,cookie-reviewers,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D251177
# ==============================================================================

diff -r e93a3c226679 -r 56cce39fb13c dom/cookiestore/CookieStoreParent.h
--- a/dom/cookiestore/CookieStoreParent.h	Tue Jun 03 13:21:06 2025 +0000
+++ b/dom/cookiestore/CookieStoreParent.h	Tue Jun 03 13:21:35 2025 +0000
@@ -34,7 +34,8 @@
   ~CookieStoreParent();
 
   mozilla::ipc::IPCResult RecvGetRequest(
-      nsIURI* aCookieURI, const OriginAttributes& aOriginAttributes,
+      NotNull<RefPtr<nsIURI>> aCookieURI,
+      const OriginAttributes& aOriginAttributes,
       const Maybe<OriginAttributes>& aPartitionedOriginAttributes,
       const bool& aThirdPartyContext, const bool& aPartitionForeign,
       const bool& aUsingStorageAccess, const bool& aIsOn3PCBExceptionList,
@@ -42,21 +43,22 @@
       const bool& aOnlyFirstMatch, GetRequestResolver&& aResolver);
 
   mozilla::ipc::IPCResult RecvSetRequest(
-      nsIURI* aCookieURI, const OriginAttributes& aOriginAttributes,
-      const bool& aThirdPartyContext, const bool& aPartitionForeign,
-      const bool& aUsingStorageAccess, const bool& aIsOn3PCBExceptionList,
-      const nsString& aName, const nsString& aValue, const bool& aSession,
-      const int64_t& aExpires, const nsString& aDomain, const nsString& aPath,
-      const int32_t& aSameSite, const bool& aPartitioned,
-      const nsID& aOperationID, SetRequestResolver&& aResolver);
+      NotNull<RefPtr<nsIURI>> aCookieURI,
+      const OriginAttributes& aOriginAttributes, const bool& aThirdPartyContext,
+      const bool& aPartitionForeign, const bool& aUsingStorageAccess,
+      const bool& aIsOn3PCBExceptionList, const nsString& aName,
+      const nsString& aValue, const bool& aSession, const int64_t& aExpires,
+      const nsString& aDomain, const nsString& aPath, const int32_t& aSameSite,
+      const bool& aPartitioned, const nsID& aOperationID,
+      SetRequestResolver&& aResolver);
 
   mozilla::ipc::IPCResult RecvDeleteRequest(
-      nsIURI* aCookieURI, const OriginAttributes& aOriginAttributes,
-      const bool& aThirdPartyContext, const bool& aPartitionForeign,
-      const bool& aUsingStorageAccess, const bool& aIsOn3PCBExceptionList,
-      const nsString& aName, const nsString& aDomain, const nsString& aPath,
-      const bool& aPartitioned, const nsID& aOperationID,
-      DeleteRequestResolver&& aResolver);
+      NotNull<RefPtr<nsIURI>> aCookieURI,
+      const OriginAttributes& aOriginAttributes, const bool& aThirdPartyContext,
+      const bool& aPartitionForeign, const bool& aUsingStorageAccess,
+      const bool& aIsOn3PCBExceptionList, const nsString& aName,
+      const nsString& aDomain, const nsString& aPath, const bool& aPartitioned,
+      const nsID& aOperationID, DeleteRequestResolver&& aResolver);
 
   mozilla::ipc::IPCResult RecvGetSubscriptionsRequest(
       const PrincipalInfo& aPrincipalInfo, const nsCString& aScopeURL,
@@ -70,7 +72,8 @@
   mozilla::ipc::IPCResult RecvClose();
 
   void GetRequestOnMainThread(
-      nsIURI* aCookieURI, const OriginAttributes& aOriginAttributes,
+      const RefPtr<nsIURI> aCookieURI,
+      const OriginAttributes& aOriginAttributes,
       const Maybe<OriginAttributes>& aPartitionedOriginAttributes,
       bool aThirdPartyContext, bool aPartitionForeign, bool aUsingStorageAccess,
       bool aIsOn3PCBExceptionList, bool aMatchName, const nsAString& aName,
@@ -79,21 +82,19 @@
 
   // Returns true if a cookie notification has been generated while completing
   // the operation.
-  bool SetRequestOnMainThread(ThreadsafeContentParentHandle* aParent,
-                              nsIURI* aCookieURI, const nsAString& aDomain,
-                              const OriginAttributes& aOriginAttributes,
-                              bool aThirdPartyContext, bool aPartitionForeign,
-                              bool aUsingStorageAccess,
-                              bool aIsOn3PCBExceptionList,
-                              const nsAString& aName, const nsAString& aValue,
-                              bool aSession, int64_t aExpires,
-                              const nsAString& aPath, int32_t aSameSite,
-                              bool aPartitioned, const nsID& aOperationID);
+  bool SetRequestOnMainThread(
+      ThreadsafeContentParentHandle* aParent, const RefPtr<nsIURI> aCookieURI,
+      const nsAString& aDomain, const OriginAttributes& aOriginAttributes,
+      bool aThirdPartyContext, bool aPartitionForeign, bool aUsingStorageAccess,
+      bool aIsOn3PCBExceptionList, const nsAString& aName,
+      const nsAString& aValue, bool aSession, int64_t aExpires,
+      const nsAString& aPath, int32_t aSameSite, bool aPartitioned,
+      const nsID& aOperationID);
 
   // Returns true if a cookie notification has been generated while completing
   // the operation.
   bool DeleteRequestOnMainThread(
-      ThreadsafeContentParentHandle* aParent, nsIURI* aCookieURI,
+      ThreadsafeContentParentHandle* aParent, const RefPtr<nsIURI> aCookieURI,
       const nsAString& aDomain, const OriginAttributes& aOriginAttributes,
       bool aThirdPartyContext, bool aPartitionForeign, bool aUsingStorageAccess,
       bool aIsOn3PCBExceptionList, const nsAString& aName,